things/api.py

```diff
@@ -677,9 +677,53 @@ def token(**kwargs) -> Union[str, None]:
     return database.get_url_scheme_auth_token()
 
 
-def link(uuid):
-    """Return a things:///show?id=uuid link."""
-    return f"things:///show?id={uuid}"
+def url(uuid=None, command="show", **kwargs) -> str:
+    """
+    Return a things:///<command>?id=uuid&… url.
+
+    For details about available commands and their parameters
+    consult the Things URL Scheme documentation
+    [here](https://culturedcode.com/things/help/url-scheme/).
+
+    Parameters
+    ----------
+    uuid : str, optional
+        A valid uuid of any Things object
+        Defaults to None
+
+    action : str, optional
+        A valid thingsURL command
+        Defaults to show
+
+    **kwargs:
+        Any aditional parameters needed.
+
+    Examples
+    --------
+    >>> things.url("test_uuid")
+    'things:///show?id=test_uuid'
+    >>> things.url(command="update", uuid="test_uuid", title="new title")
+    'things:///update?id=test_uuid&title=new%20title&auth-token=vKkylosuSuGwxrz7qcklOw'
+    >>> things.url(command="add", title="new task", when="in 3 days", deadline="in 6 days")
+    'things:///add?title=new%20task&when=in%203%20days&deadline=in%206%20days'
+    """
+    query_params = dict(**kwargs) if uuid is None else dict(id=uuid, **kwargs)
```

Comment:
If you don't want to mutate `kwargs`, then you can do `dict(kwargs)` as well. (Since `kwargs` is created afresh on every call, I wouldn't think there's any danger in mutating it directly either... or is it?)

Reply:
I am not that experienced using `kwargs`. I just copied the statement from the else case to get it to work.
There is no deeper though behind using `dict(**kwargs)`